packages/tools/lib/create-new-component/index.js

```diff
@@ -108,47 +35,102 @@ const getLibraryName = packageName => {
+		: jsFileContentTemplate(componentName, tagName, library, packageName);
+
+	fs.writeFileSync(filePaths.main, FileContentTemplate, { flag: "wx+" });
+	fs.writeFileSync(filePaths.css, "", { flag: "wx+" });
+	fs.writeFileSync(filePaths.template, "<div>Hello World</div>", { flag: "wx+" });
+
+	console.log(`Successfully generated ${filePaths.main}`);
+	console.log(`Successfully generated ${filePaths.css}`);
+	console.log(`Successfully generated ${filePaths.template}`);
+
+	// Change the color of the output
+		console.warn('\x1b[33m%s\x1b[0m', `
+	Make sure to import the component in your bundle by using:
+	import ${componentName} from "./dist/${componentName}.js";`);
 }
 
-const tagName = `ui5-${camelToKebabCase(componentName)}`;
-
-const filePaths = {
-	"js": `./src/${componentName}.js`,
-	"css": `./src/themes/${componentName}.css`,
-	"hbs": `./src/${componentName}.hbs`,
-};
-const sJsFileContentTemplate = jsFileContentTemplate(componentName);
-
-fs.writeFileSync(filePaths.js, sJsFileContentTemplate, { flag: "wx+" });
-fs.writeFileSync(filePaths.css, "", { flag: "wx+" });
-fs.writeFileSync(filePaths.hbs, "<div>Hello World</div>", { flag: "wx+" });
+// Main function
+const createWebComponent = async () => {
+	const consoleArguments = process.argv.slice(2);
+	let componentName = consoleArguments[0];
+	let tagName = consoleArguments[1];
+	let language = consoleArguments[2];
+	let isTypeScript;
+
+	if (!componentName) {
+		const response = await prompts({
+			type: "text",
+			name: "componentName",
+			message: "Please enter a component name:",
+			validate: (value) => isNameValid(value),
+		});
+		componentName = response.componentName;
+	}
 
+	if (!tagName) {
+		const response = await prompts({
+			type: "text",
+			name: "tagName",
+			message: "Please enter a tag name:",
+			validate: (value) => isTagNameValid(value),
+		});
+		tagName = response.tagName;
+	}
 
-console.log(`Successfully generated ${componentName}.js`);
-console.log(`Successfully generated ${componentName}.css`);
-console.log(`Successfully generated ${componentName}.hbs`);
+	if (!language && isTagNameValid(tagName)) {
+		const response = await prompts({
+			type: "select",
+			name: "language",
+			message: "Please select a language:",
+			choices: [
+				{
+					title: "TypeScript (recommended)",
+					value: true,
+				},
+				{
+					title: "JavaScript",
+					value: false,
+				},
+			],
+		});
+		isTypeScript = response.language;
+	} else if (language === "typescript" || language === "ts") {
+		isTypeScript = true;
+	} else {
+		isTypeScript = false;
+	}
 
-const bundleLogger = fs.createWriteStream("./bundle.common.js", {
-	flags: "a" // appending
-});
+	const packageName = getPackageName();
+	const library = getLibraryName(packageName);
 
-bundleLogger.write(`
-// TODO: Move this line in order to keep the file sorted alphabetically
-import ${componentName} from "./dist/${componentName}.js";`);
+	if (isNameValid(componentName) && isTagNameValid(tagName)) {
```

Comment:
There will be no need for validation any more at this step - the whole validation will be done above (see my first comment), and the wizards are guaranteed to produce a valid input always.